src/test/ui/async-await/issues/issue-65159.stderr

```diff
@@ -1,19 +1,19 @@
-error[E0107]: this enum takes 2 type arguments but only 1 type argument was supplied
+error[E0107]: this enum takes 2 generic arguments but 1 generic argument was supplied
   --> $DIR/issue-65159.rs:5:20
    |
 LL | async fn copy() -> Result<()>
-   |                    ^^^^^^ -- supplied 1 type argument
+   |                    ^^^^^^ -- supplied 1 generic argument
    |                    |
-   |                    expected 2 type arguments
+   |                    expected 2 generic arguments
    |
-note: enum defined here, with 2 type parameters: `T`, `E`
+note: enum defined here, with  2 generic parameters : `T`, `E`
   --> $SRC_DIR/core/src/result.rs:LL:COL
    |
 LL | pub enum Result<T, E> {
    |          ^^^^^^ -  -
-help: add missing type argument
+help: add missing generic argument
    |
-LL | async fn copy() -> Result<(), E>
+LL | async fn copy() -> Result<(), T>
    |                             ^^^
```

Comment:
@memoryruins is right, we're now using the wrong type parameter here.

Reply:
This assumes that we're missing the type for `Error`. If the argument for the first type parameter is missing and we supplied an argument for `Error`, then we're using the wrong type parameter. Unless we use type information here, I think there is no way to infer the right type parameter name. I chose to always start from the first type parameter to suggest names for consistency.

Reply:
Although I agree that it makes sense to assume that arguments that appear later in the parameter definition are missing. I'll change that.